### PR TITLE
Intersection with S.Reals shouldn't clear coeff

### PR DESCRIPTION
**Pull request by Shekharrajak**
Fixes some cases of https://github.com/sympy/sympy/issues/11174
##### From master branch :

```
In [1]: Intersection(FiniteSet(-x), S.Reals)
Out[1]: (-∞, ∞) ∩ {x}

In [2]: solveset(z**2 + exp(2*x) - sin(y),x,S.Reals)
Out[2]: 
          ⎧   ⎛ ⎛ 2         ⎞⎞⎫
(-∞, ∞) ∩ ⎨log⎝-⎝z  - sin(y)⎠⎠⎬
          ⎩                   ⎭

In [4]: r, t = symbols('r t')

In [5]: solveset(sqrt(r)*Abs(tan(t))/sqrt(tan(t)**2 + 1) + x*tan(t),x,S.Reals)
Out[5]: 
          ⎧      √r⋅│tan(t)│      ⎫
(-∞, ∞) ∩ ⎪───────────────────────⎪
          ⎨   _____________       ⎬
          ⎪  ╱    2               ⎪
          ⎩╲╱  tan (t) + 1 ⋅tan(t)⎭

```
##### From this branch :

```
In [ ]: Intersection(FiniteSet(-x), S.Reals)
Out[ ]: ℝ ∩ {-x}

In [ ]: solveset(z**2 + exp(2*x) - sin(y),x,S.Reals)
Out[ ]: 
    ⎧   ⎛   2         ⎞⎫
    ⎪log⎝- z  + sin(y)⎠⎪
ℝ ∩ ⎨──────────────────⎬
    ⎪        2         ⎪
    ⎩                  ⎭

In [ ]: r, t = symbols('r t')
In [ ]: solveset(sqrt(r)*Abs(tan(t))/sqrt(tan(t)**2 + 1) + x*tan(t),x,S.Reals)
Out[ ]: 
    ⎧     -√r⋅│tan(t)│      ⎫
ℝ ∩ ⎪───────────────────────⎪
    ⎨   _____________       ⎬
    ⎪  ╱    2               ⎪
    ⎩╲╱  tan (t) + 1 ⋅tan(t)⎭

```

If someone use `x, y, z` as Real number then `log(-(z**2 - sin(y)))` is not the correct ans. Ans should be `log(-(z**2 - sin(y)))/2`. similarly 3rd case.

### CODE REVIEW

**Comment by Shekharrajak** [comment]
Ping @aktech @asmeurer @hargup @kshitij10496 please share your thoughts on this.

**Comment by smichr** [comment]
Why shouldn't `{-x} intersect R` -> `{x} intersect R` (or the same if `x` is replaced with `x/2`)? Maybe I'm missing something obvious...

**Comment by asmeurer** [comment]
I'm unclear why it should. ℝ ∩ {-2} is {-2}, for instance. Perhaps you are confusing intersection and union?

**Comment by Shekharrajak** [comment]
since last week I am confused about it. I didn't find the ans in the internet.
Please see the these examples: 
(From the master branch)
```
In [28]: k = FiniteSet(-x/2, -x/3, x/5)

In [29]: Intersection(Interval(-oo, oo), k).subs(x,2)
Out[29]: {-1, -2/3, 2/5}

In [30]: k = FiniteSet(-x/2)

In [31]: Intersection(Interval(-oo, oo), k).subs(x,2)
Out[31]: {2}

In [32]: k = FiniteSet(-1)

In [33]: Intersection(Interval(-oo, oo), k)
Out[33]: {-1}

In [34]: k = FiniteSet(-x/2, -x/3, x/5)

In [35]: Intersection(Interval(-oo, oo), k)
Out[35]: 
          ⎧-x   -x   x⎫
(-∞, ∞) ∩ ⎨───, ───, ─⎬
          ⎩ 2    3   5⎭

In [36]: k = FiniteSet(-x/2)

In [37]: Intersection(Interval(-oo, oo), k)
Out[37]: (-∞, ∞) ∩ {x}

```

**Comment by smichr** [comment]
I think I'm confusing imageset and Intersection: `imagemap(x,x/2,Interval(1,2)) -> [2,4]` whereas `Intersection({x/2}, Interval(1,2))` is more like a *filterset* where the result is a set that contains elements in common with both initial sets. 

Does what you have added keep `Intersection({x/2}, Interval(1,2))` from becoming `Intersection([2, 4], {x})`?

**Comment by Shekharrajak** [comment]
@smichr 

> 
Does what you have added keep Intersection({x/2}, Interval(1,2)) from becoming Intersection([2, 4], {x})?

yes but only when Interval is `S.Reals`. in other cases both are same, right? means 
` Intersection({x/2}, Interval(1,2)) ` is same as `Intersection([2, 4], {x})`.

**Comment by smichr** [comment]
> both are same, right

I don't believe so now though I may have incorrectly believed so in the past. Here's why: what is true for a specific case must also be true for a general case when the general becomes specific. But the way things work now, they are not:

```python
>>> Intersection(S({S.Half}), Interval(0,2))
{1/2}
>>> Intersection(S({x/2}), Interval(0,2)).subs(x,1)  # replacing x with 1 should give the same answer
{1}
```

An intersection should give the elements that are in common between two or more sets.

**Comment by Shekharrajak** [comment]
@smichr , Thanks.
Means there should not be simplification and this [if block](https://github.com/Shekharrajak/sympy/blob/398b61977051524342aacc7921add37cbbe1376b/sympy/sets/sets.py#L1529) should be removed ?

I found a testcase [here](https://github.com/Shekharrajak/sympy/blob/398b61977051524342aacc7921add37cbbe1376b/sympy/sets/tests/test_sets.py#L1039) 
```
In [ ]: FiniteSet(-x - 1).intersect(Interval.Ropen(1, 2)) 
Out[ ]: (-3, -2] ∩ {x}
```
ans should be `[1, 2) ∩ {-x - 1}` ?

**Comment by smichr** [comment]
> should not be simplification

Yes. It seems obvious now yet I'm almost sure that I either advised or made the change that we are talking about.

I guess what I am/was thinking of was an ElementOf set, e.g. `x/2 \elof Range(3) => x \elof Range(6)` but I don't know if we even have such a set.

**Comment by asmeurer** [comment]
There is `Contains` (but it's a boolean, not a set).

**Comment by smichr** [comment]
> There is Contains (but it's a boolean, not a set).

Ahh...and that brings me back to the PR that is stalled that tried to get Relationals in canonical presentation so `Contains(x/2,Interval(1,3))` would give `2<=x<=6` instead of `1<=x/2<=3`.

**Comment by Shekharrajak** [comment]
From master branch
```
In [1]: Contains(x/2,Interval(1,3))
Out[1]: 
x       x    
─ ≥ 1 ∧ ─ ≤ 3
2       2  
```
this should be `x ≥ 2 ∧ x ≤ 6` . 
Also [one testcase](https://github.com/Shekharrajak/sympy/commit/7868de9498e3530312fb5074693f5fd92c9f07af#diff-85baa04bbf4e1dfd9128782738e45424R136) was changed but [wolfram](http://www.wolframalpha.com/input/?i=Abs(Abs(x+%2B+3)+-+a)+-+b+%3D+0+for+x) returns 4 soln so I think previous was correct(there were h1, h2, h3, h4 previously).

**Comment by asmeurer** [comment]
Let's leave Contains simplification to a separate PR. Are there any objections to this PR?

**Comment by Shekharrajak** [comment]
I have removed the `if` block which simplifies the symbolic intersection and changed some testcases. Please review once.

File: sympy/solvers/tests/test_solveset.py [hunk 1]
```diff
@@ -139,13 +139,20 @@ def ireal(x, s=S.Reals):
...
-    h2 = Intersection(Interval(-3, oo), FiniteSet(a - b - 3),
+    h3 = Intersection(Interval(-3, oo), FiniteSet(a - b - 3),
                       imageset(Lambda(n, -n + a - 3), Interval(0, oo)))
 
-    assert invert_real(Abs(Abs(x + 3) - a) - b, 0, x) == (x, Union(h1, h2))
+    h4 = Intersection(Interval(-oo, -3), FiniteSet(-a - b - 3),
+                      imageset(Lambda(n, n - a - 3), Interval(-oo, 0)))
+
+    soln = (x, Union(h1, h2, h3, h4))
+    assert invert_real(Abs(Abs(x + 3) - a) - b, 0, x) == soln
```
Comment (Shekharrajak) [inline sympy/solvers/tests/test_solveset.py]: I think according to [wolfram](http://www.wolframalpha.com/input/?i=Abs(Abs(x+%2B+3)+-+a)+-+b+%3D+0+for+x) this is correct.

File: sympy/sets/tests/test_sets.py [hunk 2]
```diff
@@ -1035,21 +1035,14 @@ def test_issue_8257():
     assert FiniteSet(-oo) + Interval(-oo, oo) == reals_plus_negativeinfinity
 
 
-def test_issue_10285():
```
Comment (Shekharrajak) [inline sympy/sets/tests/test_sets.py]: refere [this comment](https://github.com/sympy/sympy/pull/11280#issuecomment-228707293)

**Comment by smichr** [comment]
Looks good to me...but that may not be saying much since what you removed *also* looked good to me at the time. I'll let someone else give the final +1.

**Comment by asmeurer** [comment]
Looks good to me.

**Pull request merged**